src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -52,4 +93,278 @@ private long countGermplasm(UUID programId, String trialDbId) throws ApiExceptio
         List<BrAPIObservationUnit> obUnits = ouDAO.getObservationUnitsForTrialDbId(programId, trialDbId);
         return obUnits.stream().map(BrAPIObservationUnit::getGermplasmDbId).distinct().count();
     }
+
+    public DownloadFile exportObservations(
+            Program program,
+            UUID experimentId,
+            ExperimentExportQuery params) throws IOException, DoesNotExistException, ApiException {
+        StreamedFile downloadFile;
+        boolean isDataset = false;
+        List<BrAPIObservation> dataset = new ArrayList<>();
+        List<BrAPIObservationVariable> obsVars = new ArrayList<>();
+        Map<String, Map<String, Object>> rowByOUId = new HashMap<>();
+        Map<String, BrAPIStudy> studyByDbId = new HashMap<>();
+        List<String> requestedEnvNames = StringUtils.isNotBlank(params.getEnvironments()) ?
+                new ArrayList<>(Arrays.asList(params.getEnvironments().split(","))) : new ArrayList<>();
+        FileType fileType = params.getFileExtension();
+
+        // get requested environments for the experiment
+        List<BrAPIStudy> expStudies = studyDAO.getStudiesByExperimentID(experimentId, program);
+        if (!requestedEnvNames.isEmpty()) {
+            expStudies = expStudies
+                    .stream()
+                    .filter(study -> requestedEnvNames.contains(
+                            Utilities.removeProgramKeyAndUnknownAdditionalData(study.getStudyName(), program.getKey())))
+                    .collect(Collectors.toList());
+        }
+        expStudies.forEach(study -> studyByDbId.putIfAbsent(study.getStudyDbId(), study));
+
+        // get the OUs for the requested environments
+        List<BrAPIObservationUnit> ous = new ArrayList<>();
+        try {
+            for (BrAPIStudy study: expStudies) {
+                ous.addAll(ouDAO.getObservationUnitsForStudyDbId(study.getStudyDbId(), program));
+            }
+        } catch (ApiException err) {
+            log.error("Error fetching observation units for a study by its DbId" +
+                    Utilities.generateApiExceptionLogMessage(err), err);
+            err.printStackTrace();
```

Comment:
This isn't needed as you've already passed the exception object to the `log.error` method (and that method will print the stack trace)

Reply:
done

Reply:
done